test/base-json.cpp

```diff
@@ -42,7 +60,12 @@ BOOST_AUTO_TEST_CASE(encode)
 }
 )EOF");
 
-	BOOST_CHECK(JsonEncode(input, true) == output);
+	auto got(JsonEncode(input, true));
+	BOOST_CHECK_MESSAGE(output == got, "expected=" << output << "\ngot=" << got);
+
+	std::ostringstream oss;
+	JsonEncode(input, oss, true);
+	BOOST_CHECK_MESSAGE(oss.str() == output, "expected=" << output << "\ngot=" << oss.str());
```

Comment:
What about just using `BOOST_CHECK_EQUAL`? That outputs both values automatically.